### PR TITLE
Add short contribution "guide"

### PR DESCRIPTION
**Pull request by rusty-key**
Added steps I had to perform on clean repo as a short contribution guide.

Also gitignored `_opam` to support local switch setup ~and added `cmdliner` to dependencies as it was missing~.

### CODE REVIEW

File: ocaml-lsp-server.opam [hunk 1]
```diff
@@ -23,6 +23,7 @@ depends: [
   "ocamlfind" {>= "1.5.2"}
   "ocaml" {>= "4.06"}
   "dune" {>= "2.0"}
+  "cmdliner"
```
Comment (imbsky) [inline ocaml-lsp-server.opam:26]: You should edit the dune-project instead as described at the top of the file.

Comment (giltho) [inline ocaml-lsp-server.opam:26]: cmdliner is vendored here, so it's not an opam dependency
Also, note for the future, the `opam` file is generated automatically by dune. So if cmdliner was to be added as a dependency, it should be done by adding it in the `dune-package` file

File: README.md [hunk 2]
```diff
@@ -79,6 +79,28 @@ The server supports the following queries:
...
+
+```bash
+# clone repo with submodules
+git clone --recursive git@github.com:ocaml/ocaml-lsp.git
+
+# if you already clone, pull submodules
+git submodule update --init --recursive
+
+# creat local switch (or use global one)
+opam sw create . 4.09.0
```
Comment (imbsky) [inline README.md:92]: It's probably more clear to newcomers to use switch instead of sw. (This is just a suggestion, feel free to ignore it.)

Comment (giltho) [inline README.md:92]: I tried reproducing the steps, but I got this
```
opam sw create . 4.09.0
[ERROR] Compiler selection '4.09.0' is ambiguous. matching packages: { ocaml-base-compiler.4.09.0, ocaml-system.4.09.0 }
```
It depends on which system compiler you have, but let's replace this to make it explicit ?
Also, we might as well go with 4.09.1
```suggestion
opam switch create . ocaml-base-compiler.4.09.1
```

File: README.md [hunk 3]
```diff
@@ -79,6 +79,28 @@ The server supports the following queries:
...
+
+# install dependencies
+opam install .
```
Comment (imbsky) [inline README.md:95]:
```suggestion
opam install . --deps-only --with-test
```

File: README.md (hunk 2 above)
Comment (imbsky) [inline README.md:92]:
```suggestion
opam switch create . 4.09.0
```

Comment (imbsky) [reply README.md:92]: Oh, I missed it. (I deleted my suggestion about this to avoid confusion.)

**Comment by imbsky** [comment]
@giltho I missed a few things. Thank you for correcting/supplementing them!

**Comment by rusty-key** [comment]
What if I add `make init` something like:
```
init: 
	git submodule update --init --recursive
	rm -rf _opam _build
	opam switch create . ocaml-base-compiler.4.09.1
	opam install . --deps-only --with-test
```
So it's a one command bootstrapping?

**Comment by imbsky** [comment]
I think it's enough to show the instruction in README.

**Comment by rusty-key** [comment]
> I think it's enough to show the instruction in README.

Then it's ready.

For some reason I cannot re-request review, so please take a look

**Comment by giltho** [comment]
yep LGTM :)
Merging in

**Comment by giltho** [comment]
thank you !

**Pull request merged**